src/boilerplate.rs

```diff
@@ -84,21 +84,18 @@ fn get_container(doc: &Spec, tag: &str) -> Option<NodeRef> {
         return None;
     }
 
-    match doc.containers.get(tag) {
-        Some(container) => Some(container.clone()),
-        None => None,
-    }
+    doc.containers.get(tag).cloned()
 }
 
 fn get_container_or_head(doc: &Spec, tag: &str) -> Option<NodeRef> {
     if !doc.md.boilerplate.get(tag) {
         return None;
     }
 
-    match doc.containers.get(tag) {
-        Some(container) => Some(container.clone()),
-        None => Some(doc.head().clone()),
-    }
+    doc.containers
+        .get(tag)
+        .cloned()
+        .or(Some(doc.head().clone()))
```

Comment:
This should probably do `.or_else(..)`, otherwise you're calling `clone()` and such even if the container exists.